stock_request_purchase/models/stock_request.py

```diff
@@ -38,6 +39,13 @@ def _check_purchase_company_constrains(self):
                 )
             )
 
+    def action_cancel(self):
+        res = super().action_cancel()
```

Comment:
Remember to put docstrings:

```suggestion
    def action_cancel(self):
        """Propagate the cancellation to the generated purchase orders."""
        res = super().action_cancel()
```